### PR TITLE
Enable print_blob for CIs, disable for local

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/HypothesisWorks/hypothesis/issues/2451.

For https://github.com/HypothesisWorks/hypothesis/issues/2510.

### CODE REVIEW

**Comment by Zac-HD** [comment]
Hey @hugovk! Please:

- add yourself to the alphabetical list of authors in CONTRIBUTING.rst
- add a RELEASE.rst file - check the other PRs for examples.
- add `show_default=False` for the setting, and explain in the docstring that the default is True in CI or False in local development

**Comment by hugovk** [comment]
Updated!

And btw I checked for an undocumented `CI` env var in Azure Pipelines, but not found. Maybe they'll add it one day: GitHub Actions only recently added `CI` (https://github.com/python-pillow/Pillow/pull/4562).

**Comment by hugovk** [comment]
Ha, I must have hit a special small window: I added `RELEASE.rst` when there was one in `master`, then fixed the conflict, but I guess the release bot deletes that so got a conflict again!

Rebased!

**Comment by Zac-HD** [comment]
- rebase on master (so you're creating a minor release file)
- specify in the docstring exactly how the detection works ("True if the X or Y env vars are set, False otherwise", and use Sphinx's envvar role for bonus points)
- look at the failing tests and change the assertions to account for is_in_ci().

And then I think we'll be done!

**Comment by hugovk** [comment]
> * specify in the docstring exactly how the detection works ("True if the X or Y env vars are set, False otherwise", and use Sphinx's envvar role for bonus points)

By docstring, do you mean in `description=` of `settings._define_setting`, or to add a proper `"""..."""` one to `is_in_ci()`?

**Comment by Zac-HD** [comment]
The value of the description argument, which is eventually formatted into a docstring and rendered in our public docs. Sorry for any confusion!

**Comment by hugovk** [comment]
Okay! How's that? 

> * look at the failing tests and change the assertions to account for is_in_ci().

Hmm, do we want those tests to call `is_in_ci()` to decide what to expect? If so, should we move `is_in_ci()` to some utility file?

**Comment by Zac-HD** [comment]
Yes, but let's leave it in _settings.py near its only use in the non-test code.

**Comment by Zac-HD** [comment]
Hmm.  Looking at the actual change, I think it's actually going to be cleaner if we just add `print_blob=False` to the settings decorator on those inner test functions (or add the decorator too, in one case).  Sorry I didn't pick this up before - I think we're otherwise ready to merge!

File: hypothesis-python/src/hypothesis/_settings.py [hunk 1]
```diff
@@ -609,13 +610,25 @@ def _validate_deadline(x):
...
 
 
+def is_in_ci() -> bool:
+    # GitHub Actions, Travis CI and AppVeyor have "CI"
+    # Azure Pipelines has "TF_BUILD"
+    return "CI" in os.environ or "TF_BUILD" in os.environ
+
+"""
+
+"""
```
Comment (Zac-HD) [inline hypothesis-python/src/hypothesis/_settings.py:620]: Debugging leftovers?  Otherwise looks great!

Comment (hugovk) [reply hypothesis-python/src/hypothesis/_settings.py]: Oops! Removed!

**Comment by hugovk** [comment]
The docs build complains about:

```
/home/vsts/work/1/s/hypothesis-python/docs/../src/hypothesis/__init__.py:docstring of hypothesis.settings.print_blob:1: WARNING: 'envvar' reference target not found: CI
/home/vsts/work/1/s/hypothesis-python/docs/../src/hypothesis/__init__.py:docstring of hypothesis.settings.print_blob:1: WARNING: 'envvar' reference target not found: TF_BUILD
```

https://dev.azure.com/HypothesisWorks/Hypothesis/_build/results?buildId=4110&view=logs&j=6fe85d43-0322-5a7e-7a5c-a08b11a8fdaa&t=7b5ce327-7af2-52df-14b0-47f2945e41c0

**Pull request merged**